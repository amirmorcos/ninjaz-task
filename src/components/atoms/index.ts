import Icon from "./Icon/Icon";
import Divider from "./Divider/Divider";
import Avatar from "./Avatar/Avatar";

export { Icon, Divider, Avatar };
