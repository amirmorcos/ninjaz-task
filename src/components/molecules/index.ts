import AppImage from "./AppImage/AppImage";
import PostItem from "./PostItem/PostItem";

export { AppImage, PostItem };
