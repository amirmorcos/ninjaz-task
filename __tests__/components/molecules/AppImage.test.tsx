import React from "react";
import renderer from "react-test-renderer";
import { AppImage } from "../../../src/components/molecules";

jest.mock("react-native-vector-icons/Ionicons", () => "Icon");
jest.mock("react-native-toast-message", () => ({
  show: jest.fn(),
  hide: jest.fn(),
}));
jest.mock("react-native-size-matters", () => ({
  scale: () => {},
  verticalScale: () => {},
  moderateScale: () => {},
}));

test("renders correctly", () => {
  const tree = renderer
    .create(
      <AppImage image="https://randomuser.me/api/portraits/med/women/33.jpg" />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("renders placeholder correctly", () => {
  const tree = renderer.create(<AppImage image="" />).toJSON();
  expect(tree).toMatchSnapshot();
});
