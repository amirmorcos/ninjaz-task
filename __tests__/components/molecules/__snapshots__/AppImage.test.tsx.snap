// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    [
      {
        "overflow": "hidden",
      },
      [
        {
          "borderRadius": undefined,
          "height": undefined,
          "marginVertical": undefined,
          "width": undefined,
        },
        undefined,
      ],
    ]
  }
>
  <FastImageView
    defaultSource={null}
    resizeMode="cover"
    style={
      {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
</View>
`;

exports[`renders placeholder correctly 1`] = `
<View
  style={
    [
      {
        "overflow": "hidden",
      },
      [
        {
          "borderRadius": undefined,
          "height": undefined,
          "marginVertical": undefined,
          "width": undefined,
        },
        undefined,
      ],
    ]
  }
>
  <FastImageView
    defaultSource={null}
    resizeMode="cover"
    style={
      {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
</View>
`;
